kong/cache.lua

```diff
@@ -31,6 +31,28 @@ local function log(lvl, ...)
   return ngx_log(lvl, "[DB cache] ", ...)
 end
 
+-- Temporary fix to convert soft callback errors into hard ones.
+-- FIXME: use upstream mlcache lib instead of local copy
+local soft_to_hard
+do
+  local s2h_cache = setmetatable({}, { __mode = "k" })
+  
+  local function create_wrapper(cb)
+    s2h_cache[cb] = function(...)
+      local result, err = cb(...)
+      if result == nil and err then
```

Comment:
It is instructed to users of this API that they should never rely on the truthiness/falseness of the first returned value to infer errors (since the library caches misses as well). As such, as a developer I would easily expect that when I write a callback, I can return whichever value I want when the second one is an error (non-nil). While this isn't the typical idiom we try to follow in this codebase (we prefer `nil` + `err`), I believe this comes down to usability/robustness: we should only check for non-nil errors here to infer one (disregarding the first return value).

Reply:
@thibaultcha all comments addressed

Reply:
Looking good! Let's wait for the tests to double check.